dnf5/context.cpp

```diff
@@ -87,6 +87,35 @@ class KeyImportRepoCB : public libdnf5::repo::RepoCallbacks {
         std::cerr << _("The key was successfully imported.") << std::endl;
     }
 
+    bool repokey_remove(const libdnf5::rpm::KeyInfo & key_info) override {
+        if (config->get_assumeno_option().get_value()) {
+            return false;
+        }
+
+        std::cerr << libdnf5::utils::sformat(
+                         _("The following PGP key (0x{}) has expired on {}:"),
```

Comment:
The `repokey_remove` callback is called before the key is removed and allows you to confirm or deny its removal.

The message "The following PGP key (0x{}) expired on {}:" may be wrong. We do not know who called the callback and why they want to remove the key. It could was the expired-pgp-keys plugin. But also some other plugin, which for example will check if the key has been revoked. Or whether the key is still on the company's list of allowed keys. The reason for removing the key could be different.